### ISSUE TITLE
Show only one pending/failed tests summary for multiple run of RSpec::Core::Runner#run 

### ISSUE THREAD

**Issue by ArturT** [rspec/rspec-core#2416]
I run multiple times `RSpec::Core::Runner#run` to execute test files. For each run the pending test files and failed test files are shown in the rspec summary. So the second run `RSpec::Core::Runner#run` shows again the pending/failed tests from first run `RSpec::Core::Runner#run`.

Is there a way to show pending/failed tests only once? I was wondering if it would be possible to don't show it for each run `RSpec::Core::Runner#run` and run some rspec command at the very end to display rspec summary with all pending/failed test files without running any tests instead of showing repeated summary for each intermediate run `RSpec::Core::Runner#run`. It would be great if you could share tip what could I do. Thanks!

[Source file](https://github.com/KnapsackPro/knapsack_pro-ruby/blob/master/lib/knapsack_pro/runners/queue/rspec_runner.rb#L42).
```ruby
# fetch some test files from API and add to cli_args
options = RSpec::Core::ConfigurationOptions.new(cli_args)
exit_code = RSpec::Core::Runner.new(options).run($stderr, $stdout)
RSpec.world.example_groups.clear

# fetch more test files from API and set a newd cli_args with those test files
options = RSpec::Core::ConfigurationOptions.new(cli_args)
exit_code = RSpec::Core::Runner.new(options).run($stderr, $stdout)
RSpec.world.example_groups.clear

# when API returns no test files then do nothing
```

Example rspec output with repeated pending tests for second run:

```
################### First run

Calculator
  behaves like calculator
    behaves like has add method
      should equal true
    behaves like has mal method
      should equal true
  #add
    should eq 5
  #mal
    should eq 6

Meme
  #i_can_has_cheezburger?
    should eq "OHAI!"
  #will_it_blend?
    should eq "YES!"

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Calculator when add two numbers this must fail
     # Temporarily skipped with xit
     # ./spec/features/calculator_spec.rb:30


Finished in 0.02665 seconds (files took 3.26 seconds to load)
15 examples, 0 failures, 1 pending


################### Second run

Bar
  example at ./spec/bar_spec.rb:2 (PENDING: Temporarily skipped with xit)

Timecop
  should eq 21

Foo
  should equal true

FooBar
  should equal true

Pending: (Failures listed here are expected and do not affect your suite's status)

  1) Calculator when add two numbers this must fail
     # Temporarily skipped with xit
     # ./spec/features/calculator_spec.rb:30

  2) Bar
     # Temporarily skipped with xit
     # ./spec/bar_spec.rb:2


Finished in 0.03462 seconds (files took 4.15 seconds to load)
19 examples, 0 failures, 2 pending

```

**Comment by JonRowe** [comment]
Yes, you can write a custom formatter that doesn't output the summary, see https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/formatters/protocol.rb for documentation on how to implement one.

**Comment by xaviershay** [comment]
Also I can't remember the exact differences but `RSpec.clear_examples` might be a better option. I fixed some related things with it recently: https://github.com/rspec/rspec-core/pull/2368

**Comment by ArturT** [comment]
I wrote my custom formatter. I also tried [fix #2368](https://github.com/rspec/rspec-core/pull/2368) `RSpec.clear_examples` but it's on master and not yet released and it turned out I prefer to keep a cumulative summary of pending/failed tests so my customer format can show it at the very end and it works with old rspec and with the new one when you will release 3.6.

Thanks for help!
